test/controllers/basic_search_controller_test.rb

```diff
@@ -6,6 +6,48 @@ class BasicSearchControllerTest < ActionDispatch::IntegrationTest
     assert_response :success
 
     assert_select 'form#basic-search', { count: 1 }
+
+    details_div = assert_select('details#advanced-search-panel')
+    assert_nil details_div.attribute('open')
+  end
+
+  test 'advanced search form appears on index with URL parameter' do
+    get '/?advanced=true'
+
+    assert_response :success
+
+    details_div = assert_select('details#advanced-search-panel')
+    refute_nil details_div.attribute('open')
```

Comment:
If it works, I might use `assert` instead of `refute_nil` here to avoid the cognitive load of processing double negatives.